docs/groovy-script/mods/abyssaltweaker/materialization.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "Materialization"
+titleTemplate: "Abyssal Tweaker | CleanroomMC"
+description: "Turns Crystals from a Crystal Bag into an output item."
+source_code_link: "https://github.com/TeamDimensional/AbyssalTweaker/blob/master/src/main/java/com/teamdimensional/abyssaltweaker/compat/groovyscript/RegistryMaterialization.java"
+---
+
+# Materialization (Abyssal Tweaker)
+
+## Description
+
+Turns Crystals from a Crystal Bag into an output item.
+
+:::::::::: details Warning {open id="warning"}
+The Materializer compatibility will only take the first item that satisfies each ingredient into account. This is fine if ItemIngredients are used, but will not work with OrIngredients and the like.
```

Comment:
i dont understand what this means. could this be clarified in the documentation?

Reply:
I'm not sure how to describe this properly, but basically this compatibility does `getMatchingItems().get(0)` to determine the input item. would not mind a better explanation (not native english etc etc)

Reply:
i think this should have a `required` element added to the `input` `@Property` annotation - and perhaps a better way to word it might be 
> The Materializer Recipe Builder must have all inputs be an AbyssalCraft Crystal. This is checked via looking at the first entry of `getMatchingStacks`, meaning that the `IIngredient` used should be obtained via the `item` Object Mapper to ensure it is valid.